static/src/javascripts/projects/common/modules/identity/api.js

```diff
@@ -0,0 +1,284 @@
+// @flow
+
+/* global escape:true */
+import { ajax } from 'lib/ajax';
+import config from 'lib/config';
+import { getCookie as getCookieByName } from 'lib/cookies';
+import mediator from 'lib/mediator';
+import { local } from 'lib/storage';
+import { mergeCalls } from 'common/modules/asyncCallMerger';
+
+/**
+ * Left this as an object as there are only static methods
+ * We'll need to change this once there is some state change
+ * TODO(jamesgorrie): Allow this to show policies too (not needed yet)
+ */
+
+let userFromCookieCache = null;
+
+const Id = {
+    cookieName: 'GU_U',
+    signOutCookieName: 'GU_SO',
+    fbCheckKey: 'gu.id.nextFbCheck',
+    lastRefreshKey: 'identity.lastRefresh',
+    idApiRoot: null,
+};
+
+export const init = () => {
+    Id.idApiRoot = config.page.idApiUrl;
+    mediator.emit('module:identity:api:loaded');
+};
+
+/**
+ * Handles unicode chars correctly
+ * @param {string} str
+ * @return {string}
+ */
+export const decodeBase64 = (str: string) =>
+    decodeURIComponent(
+        escape(
+            window.atob(
+                str
+                    .replace(/-/g, '+')
+                    .replace(/_/g, '/')
+                    .replace(/,/g, '=')
+            )
+        )
+    );
+
+/**
+ * The object returned from the cookie has the keys:
+ *
+ * {
+ *    id
+ *    primaryEmailAddress
+ *    displayName
+ *    accountCreatedDate
+ *    emailVerified
+ *    rawResponse
+ * };
+ *
+ * @return {?Object} the user information
+ */
+export const getUserFromCookie = (): ?Object => {
+    if (userFromCookieCache === null) {
+        const cookieData = getCookieByName(Id.cookieName);
+        let userData = null;
+
+        if (cookieData) {
+            userData = JSON.parse(decodeBase64(cookieData.split('.')[0]));
+        }
+        if (userData) {
+            const displayName = decodeURIComponent(userData[2]);
+            userFromCookieCache = {
+                id: userData[0],
+                primaryEmailAddress: userData[1], // not sure where this is stored now - not in the cookie any more
+                displayName,
+                accountCreatedDate: userData[6],
+                emailVerified: userData[7],
+                rawResponse: cookieData,
+            };
+        }
+    }
+
+    return userFromCookieCache;
+};
+
+/**
+ * @return {boolean}
+ */
+export const isUserLoggedIn = () => getUserFromCookie() !== null;
+
+/**
+ * Gets the currently logged in user data from the identity api
+ * @param {function} callback
+ */
+export const getUserFromApi = mergeCalls(mergingCallback => {
+    const apiRoot = Id.idApiRoot ? Id.idApiRoot : '';
```

Comment:
You could remove the ternary operation, eg. `const apiRoot = Id.idApiRoot || '' `

Reply:
Fixed in 01f1c9c9f084850fc7563854ded81a4c8f3c493c